src/test/java/org/zalando/riptide/RouterTest.java

```diff
@@ -47,26 +50,62 @@
     @Rule
     public final ExpectedException exception = ExpectedException.none();
 
-    private final Router unit = new Router();
+    @Test
+    public void shouldCreateNewRouterIfChanged() {
+        Router<HttpStatus> router = Router.create(status(), on(OK).capture());
+        Router<HttpStatus> result = router.add(anyStatus().capture());
+        Assert.assertNotEquals(router, result);
+    }
+
+    @Test
+    public void shouldCreateNewRouterIfNotChanged() {
+        Router<HttpStatus> router = Router.create(status(), on(OK).capture());
+        Router<HttpStatus> result = router.add(on(OK).capture());
+        Assert.assertNotEquals(router, result);
+    }
+
+    @Test
+    public void shouldSubstituteBinding() {
```

Comment:
> Would be nice to have the same test for wildcards, since they are implemented as null I'd assume it could get tricky there.

Not from code perspective, but you are right from behavior perspective.